lib/vssh/libssh.c

```diff
@@ -2570,7 +2579,39 @@ static ssize_t sftp_send(struct Curl_easy *data, int sockindex,
    */
   if(len > 32768)
     len = 32768;
-
+#if LIBSSH_VERSION_INT > SSH_VERSION_INT(0, 11, 0)
+  switch(conn->proto.sshc.sftp_send_state) {
+    case 0:
+      sftp_file_set_nonblocking(conn->proto.sshc.sftp_file);
+      if(sftp_aio_begin_write(conn->proto.sshc.sftp_file, mem, len,
+                              &conn->proto.sshc.sftp_aio) == SSH_ERROR) {
+        *err = CURLE_SEND_ERROR;
+        return -1;
+      }
+      FALLTHROUGH();
+    case 1:
+      conn->proto.sshc.sftp_send_state = 1;
+      nwrite = sftp_aio_wait_write(&conn->proto.sshc.sftp_aio);
```

Comment:
```suggestion
      }
      conn->proto.sshc.sftp_send_state = 1;
      FALLTHROUGH();
    case 1:
      nwrite = sftp_aio_wait_write(&conn->proto.sshc.sftp_aio);
```
do not update `conn→proto.sshc.sftp_send_state` to 1,
if it already has the value?

Reply:
Yes, have moved it before FALLTHROUGH.

Reply:
Thanks you @zuoxiaofeng!